aspnetcore/fundamentals/middleware/rate-limit/samples/WebRateLimitAuth/Controllers/HomeController.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Diagnostics;
+using WebRateLimitAuth.Models;
+
+namespace WebRateLimitAuth.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly ILogger<HomeController> _logger;
+
+        public HomeController(ILogger<HomeController> logger)
+        {
+            _logger = logger;
+        }
+
+        public IActionResult Index()
+        {
+            return View();
+        }
+
+        public async Task<IActionResult> Privacy()
+        {
+            await Task.Delay(100);
+            return View();
+        }
```

Comment:
The MVC controller only uses the global rate limiter. Using Azure load testing with a JMeter jmx script, I start getting 503 errors pretty quickly.

It looks like you can't apply the rate controller to specific actions, controllers, or razor pages. Is that right?

Reply:
I tried to test this in chrome by refreshing all tabs but chrome must use a single thread or something because even with 50 tabs all refreshing I couldn't get a 503 error and each tab would show busy. They eventually all responded with no errors.

Reply:
> The MVC controller only uses the global rate limiter.

Right now there's no `RequireRateLimiting` on `MapDefaultControllerRoute`. 
https://github.com/dotnet/AspNetCore.Docs/blob/e6dbddea9c9765b06b2026596f903bc89801bd47/aspnetcore/fundamentals/middleware/rate-limit/samples/WebRateLimitAuth/Program.cs#L111

> It looks like you can't apply the rate controller to specific actions, controllers, or razor pages. Is that right?

You will be able to do it after https://github.com/dotnet/aspnetcore/pull/43053 get merged.

>  tried to test this in chrome by refreshing all tabs but chrome must use a single thread or something because even with 50 tabs all refreshing I couldn't get a 503 error and each tab would show busy.

Right now there's no limiter with loopback IP. I'm guessing maybe you are calling from 127.0.0.1 🤔

Reply:
You're right. Now I get the timeout:
```

    //if (!IPAddress.IsLoopback(remoteIPaddress!))
    //{
        return RateLimitPartition.CreateTokenBucketLimiter<IPAddress>
           (remoteIPaddress!, key =>
                 new TokenBucketRateLimiterOptions(tokenLimit: 5,
                     queueProcessingOrder: QueueProcessingOrder.OldestFirst,
                     queueLimit: 1,
                     replenishmentPeriod: TimeSpan.FromSeconds(15),
                     tokensPerPeriod: 1,                    
                     autoReplenishment: true));
    //}
    //else
    //{
    //    return RateLimitPartition.CreateNoLimiter<IPAddress>(IPAddress.Loopback);
    //}
```